exchange/bidder.go

```diff
@@ -392,13 +395,26 @@ func (bidder *bidderAdapter) requestBid(ctx context.Context, bidderRequest Bidde
 			}
 		} else {
 			errs = append(errs, httpInfo.err)
+			errorCode := errortypes.ReadCode(httpInfo.err)
+			nonBidReason := ErrorToNonBidReason(errorCode)
+			seatNonBids = &openrtb_ext.SeatNonBid{
+				Seat:   string(bidderRequest.BidderName),
+				NonBid: make([]openrtb_ext.NonBid, len(bidderRequest.BidRequest.Imp)),
+			}
+			for i, imp := range bidderRequest.BidRequest.Imp {
```

Comment:
@SyntaxNode  and team:
If you look at the above code snippet, we are applying `nonBidReason` flat across all impressions here. As, we are not able to identify per impression level scenario here.

Reply:
@SyntaxNode / @hhhjort : Can you update here?

Reply:
Isn't this address by https://github.com/prebid/prebid-server/issues/2973?

Reply:
@SyntaxNode : #2973 will address adapter level changes. Until, adapters are not providing impression level information in `RequestData`, this block of implementation will apply `nonBidReason` flat across all impressions.

So, I am proposing, till #2973 not effective, can we move on with the above assumption. I know there will be ambiguous cases, where adapter may have bided for the impression but still will add it under non bid because other impression may have timed out

Reply:
We can discuss at our next catchup. I was under the impression we decided to pause this issue until we implement #2973. Then we can avoid ambiguous cases.